common/apps/apps.go

```diff
@@ -57,6 +63,35 @@ func (al *Apps) Last() *App {
 	return &al.apps[len(al.apps)-1]
 }
 
+// Validate validates al for things like referential integrity of mounts<->volumes.
+func (al *Apps) Validate() error {
+	vs := map[types.ACName]struct{}{}
```

Comment:
I was about to tell you something about using `var vs map[types.ACName]struct{} = nil` to avoid memory allocation, but apparently it works only for slices. +1 point for Go for consistency.